js/core/template_manager.js

```diff
@@ -0,0 +1,277 @@
+        )[0];
+        if(cachedTemplate) return cachedTemplate.template;
+
+        const template = this._createTemplate(templateSource);
+        this._tempTemplates.push({ template, source: TemplateManager.templateKey(templateSource) });
+        return template;
+    }
+
+    _createTemplate(templateSource) {
+        return this._createElement(TemplateManager.validateTemplateSource(templateSource));
+    }
+
+    getTemplate(templateSource, templates, { isAsyncTemplate, skipTemplates }, context) {
+        if(isFunction(templateSource)) {
+            return new FunctionTemplate((options) => {
+                const templateSourceResult = templateSource.apply(context, TemplateManager._getNormalizedTemplateArgs(options));
+                // const templateSourceResult = templateSource(TemplateManager._getNormalizedTemplateArgs(options)); // accordion tests
+
+                if(!isDefined(templateSourceResult)) {
+                    return new EmptyTemplate();
+                }
+
+                let dispose = false;
+                const template = this._acquireTemplate(templateSourceResult, (templateSource) => {
+                    if(templateSource.nodeType || isRenderer(templateSource) && !$(templateSource).is('script')) {
+                        return new FunctionTemplate(() => templateSource);
+                    }
+                    dispose = true;
+                    return this._createTemplate(templateSource);
+                }, templates, isAsyncTemplate, skipTemplates);
+
+                const result = template.render(options);
+                dispose && template.dispose && template.dispose();
+                return result;
+            });
+        }
+
+        return this._acquireTemplate(templateSource, this._createTemplateIfNeeded.bind(this), templates, isAsyncTemplate, skipTemplates);
+    }
+
+    _acquireTemplate(templateSource, createTemplate, templates, isAsyncTemplate, skipTemplates) {
+        if(templateSource == null) {
+            return new EmptyTemplate();
+        }
+
+        if(templateSource instanceof ChildDefaultTemplate) {
+            return this._defaultTemplates[templateSource.name];
+        }
+
+        if(templateSource instanceof TemplateBase) {
+            return templateSource;
+        }
+
+        // TODO: templateSource.render is needed for angular2 integration. Try to remove it after supporting TypeScript modules.
+        if(isFunction(templateSource.render) && !isRenderer(templateSource)) {
+            return TemplateManager._addOneRenderedCall(templateSource);
+        }
+
+        if(templateSource.nodeType || isRenderer(templateSource)) {
+            return createTemplate($(templateSource));
+        }
+
+        return this._acquireStringTemplate(templateSource, createTemplate, templates, isAsyncTemplate, skipTemplates);
+    }
+
+    _renderIntegrationTemplate(templateSource, templates, isAsyncTemplate) {
```

Comment:
```suggestion
    _renderIntegrationTemplate(integrationTemplate, isAsyncTemplate) {
```
as `templateSource` and `templates` are used only for `const integrationTemplate = templates[templateSource];`